rust/src/ikev1/parser.rs

```diff
@@ -0,0 +1,732 @@
+                                24 => AttributeValue::GroupModp2048With256BitPrime,
+                                25 => AttributeValue::GroupRandomEcp192,
+                                26 => AttributeValue::GroupRandomEcp224,
+                                27 => AttributeValue::GroupBrainpoolEcp224,
+                                28 => AttributeValue::GroupBrainpoolEcp256,
+                                29 => AttributeValue::GroupBrainpoolEcp384,
+                                30 => AttributeValue::GroupBrainpoolEcp512,
+                                _ => AttributeValue::Unknown
+                            }
+                            11 => match attribute_length_or_value {
+                                1 => AttributeValue::LifeTypeSeconds,
+                                2 => AttributeValue::LifeTypeKilobytes,
+                                _ => AttributeValue::Unknown
+                            }
+                            _ => AttributeValue::Unknown
+                        },
+                        numeric_value: match format.0 {
+                            1 => Some(attribute_length_or_value as u32),
+                            0 => {
+                                if let Some(_numeric_variable_value) = numeric_variable_value {
+                                    Some(_numeric_variable_value)
+                                }
+                                else {
+                                    None
+                                }
+                            },
+                            _ => None,
+                        },
+                        hex_value: match format.0 {
+                            0 => {
+                                if let Some(_variable_attribute_value) = variable_attribute_value {
+                                    Some(to_hex(_variable_attribute_value))
+                                } else {
+                                    None
+                                }
+                            }
+                            _ => None,
+                        }
+                    }
+                )
+            )
+        )
+    )
+}
+
+pub fn parse_nonce(
+    i: &[u8],
+    length: u16,
+) -> IResult<&[u8], NoncePayload> {
+    do_parse!(
+        i,
+        nonce_data: take!(length) >>
+        (
+            NoncePayload {
+                nonce_data: nonce_data
+            }
+        )
+    )
+}
+
+
+named! { pub parse_ikev1_payload_list<&[u8], Vec<IsakmpPayload>>,
+    many0!(
+        complete!(
+            do_parse!(
+                next_payload: be_u8 >>
+                reserved: be_u8 >>
+                payload_length: be_u16 >>
+                data: take!(payload_length - 4) >>
+                (
+                    IsakmpPayload {
+                        payload_header: IsakmpPayloadHeader {
+                            next_payload: next_payload,
```

Comment:
Please use Field Init Shorthand (https://doc.rust-lang.org/book/ch05-01-defining-structs.html#using-the-field-init-shorthand-when-variables-and-fields-have-the-same-name)